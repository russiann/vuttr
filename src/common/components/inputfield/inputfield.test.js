import React from 'react';
import {render, cleanup} from '@testing-library/react';

import InputField from './inputfield.component';

afterEach(cleanup);

it('snapshot', () => {
  const {asFragment} = render(
    <InputField
      type="text"
      label="Tool Name"
      placeholder="lorem ipsum dolor sit amet..."
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should not show label if it does not exists', () => {
  const {asFragment, queryByTestId} = render(
    <InputField type="text" placeholder="lorem ipsum dolor sit amet..." />
  );
  const label = queryByTestId('label');
  expect(asFragment()).toMatchSnapshot();
  expect(label).toBe(null);
});

it('should show label if it exists', () => {
  const labelText = 'Tool Name';
  const {asFragment, getByTestId} = render(
    <InputField
      type="text"
      label={labelText}
      placeholder="lorem ipsum dolor sit amet..."
    />
  );
  const label = getByTestId('label');
  expect(asFragment()).toMatchSnapshot();
  expect(label).toHaveTextContent(labelText);
});

it('should not show errorMessage if it does not exists', () => {
  const {asFragment, queryByTestId} = render(
    <InputField type="text" placeholder="lorem ipsum dolor sit amet..." />
  );
  const errorMessage = queryByTestId('error-message');
  expect(asFragment()).toMatchSnapshot();
  expect(errorMessage).toBe(null);
});

it('should show errorMessage if it exists', () => {
  const errorMessageText = 'Invalid!';
  const {asFragment, getByTestId} = render(
    <InputField
      type="text"
      placeholder="lorem ipsum dolor sit amet..."
      errorMessage={errorMessageText}
    />
  );
  const errorMessage = getByTestId('error-message');
  expect(asFragment()).toMatchSnapshot();
  expect(errorMessage).toHaveTextContent(errorMessageText);
});
