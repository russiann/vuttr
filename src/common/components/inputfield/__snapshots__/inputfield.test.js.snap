// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not show errorMessage if it does not exists 1`] = `
<DocumentFragment>
  .emotion-2 {
  margin-bottom: 15px;
}

.emotion-0 {
  font-family: 'Ubuntu',sans-serif;
  outline: none;
  padding: 10px 15px;
  border: 1px solid #bdbdbd;
  border-radius: 30px;
  min-width: 200px;
  box-sizing: border-box;
  width: 100%;
  font-size: 12px;
}

<div
    class="emotion-2 emotion-3"
  >
    <input
      class="emotion-0 emotion-1"
      data-testid="textfield"
      placeholder="lorem ipsum dolor sit amet..."
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`should not show label if it does not exists 1`] = `
<DocumentFragment>
  .emotion-2 {
  margin-bottom: 15px;
}

.emotion-0 {
  font-family: 'Ubuntu',sans-serif;
  outline: none;
  padding: 10px 15px;
  border: 1px solid #bdbdbd;
  border-radius: 30px;
  min-width: 200px;
  box-sizing: border-box;
  width: 100%;
  font-size: 12px;
}

<div
    class="emotion-2 emotion-3"
  >
    <input
      class="emotion-0 emotion-1"
      data-testid="textfield"
      placeholder="lorem ipsum dolor sit amet..."
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`should show errorMessage if it exists 1`] = `
<DocumentFragment>
  .emotion-4 {
  margin-bottom: 15px;
}

.emotion-0 {
  font-family: 'Ubuntu',sans-serif;
  outline: none;
  padding: 10px 15px;
  border: 1px solid #bdbdbd;
  border-radius: 30px;
  min-width: 200px;
  box-sizing: border-box;
  width: 100%;
  font-size: 12px;
}

.emotion-2 {
  font-size: 12px;
  margin-right: 8px;
  margin-top: 4px;
  color: #ffffff;
  background: red;
  padding: 1px 2px;
  float: right;
  border-radius: 2px;
}

<div
    class="emotion-4 emotion-5"
  >
    <input
      class="emotion-0 emotion-1"
      data-testid="textfield"
      placeholder="lorem ipsum dolor sit amet..."
      type="text"
    />
    <div
      class="emotion-2 emotion-3"
      data-testid="error-message"
    >
      Invalid!
    </div>
  </div>
</DocumentFragment>
`;

exports[`should show label if it exists 1`] = `
<DocumentFragment>
  .emotion-4 {
  margin-bottom: 15px;
}

.emotion-0 {
  color: ff6007;
  font-size: 14px;
  margin-bottom: 5px;
  margin-left: 7.5px;
}

.emotion-2 {
  font-family: 'Ubuntu',sans-serif;
  outline: none;
  padding: 10px 15px;
  border: 1px solid #bdbdbd;
  border-radius: 30px;
  min-width: 200px;
  box-sizing: border-box;
  width: 100%;
  font-size: 12px;
}

<div
    class="emotion-4 emotion-5"
  >
    <div
      class="emotion-0 emotion-1"
      data-testid="label"
    >
      Tool Name
    </div>
    <input
      class="emotion-2 emotion-3"
      data-testid="textfield"
      placeholder="lorem ipsum dolor sit amet..."
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`snapshot 1`] = `
<DocumentFragment>
  .emotion-4 {
  margin-bottom: 15px;
}

.emotion-0 {
  color: ff6007;
  font-size: 14px;
  margin-bottom: 5px;
  margin-left: 7.5px;
}

.emotion-2 {
  font-family: 'Ubuntu',sans-serif;
  outline: none;
  padding: 10px 15px;
  border: 1px solid #bdbdbd;
  border-radius: 30px;
  min-width: 200px;
  box-sizing: border-box;
  width: 100%;
  font-size: 12px;
}

<div
    class="emotion-4 emotion-5"
  >
    <div
      class="emotion-0 emotion-1"
      data-testid="label"
    >
      Tool Name
    </div>
    <input
      class="emotion-2 emotion-3"
      data-testid="textfield"
      placeholder="lorem ipsum dolor sit amet..."
      type="text"
    />
  </div>
</DocumentFragment>
`;
